aisynphys/pipeline/multipatch/morphology.py

```diff
@@ -157,15 +159,19 @@ def ready_jobs(self):
             ready[expt_id] = {'dep_time': expt_mtime}
             cell_hash_compare = []
             for cell_ext_id, cell in expt.cells.items():
-                if cell.morphology is None:
+                cell_specimen_id = cell.meta.get('lims_specimen_id')
+                if cell.morphology is None or cell_specimen_id is None:
                     cell_hash_compare.append(False)
                     break
                 else:
-                    morpho_rec = morpho_results.get(cell.meta['lims_specimen_id'], None)
-                    if morpho_rec is None:
+                    morpho_rec = morpho_results.get(cell_specimen_id, None)
+                    cortical_layer = lims_layers.get(cell_specimen_id)
+                    if cortical_layer is not None:
+                        cortical_layer = cortical_layer.lstrip('Layer')
+                    if morpho_rec is None and cortical_layer is None:
                         morpho_db_hash = None
                     else:
-                        morpho_db_hash = hash_record(morpho_rec)
+                        morpho_db_hash = hash_record([morpho_rec, cortical_layer])
                     prev_hash = None if cell.morphology.meta is None else cell.morphology.meta['morpho_db_hash']
                     cell_hash_compare.append(morpho_db_hash==prev_hash)
                     # assert expt.ext_id!='1523916429.198' or cell_hash_compare[-1]
```

Comment:
I think this can be simplified:

    needs_update = False
    for cell_ext_id, cell in expt.cells.items():
        if cell.morphology is None:
            needs_update = True
            break
        cell_specimen_id = cell.meta.get('lims_specimen_id')
        morpho_rec = morpho_results.get(cell_specimen_id, None)
        cortical_layer = lims_layers.get(cell_specimen_id, None)
        morpho_db_hash = hash_record([morpho_rec, cortical_layer])
        prev_hash = None if cell.morphology.meta is None else cell.morphology.meta['morpho_db_hash']
        if morpho_db_hash != prev_hash:
            needs_update = True
            break

Reply:
(and then `if needs_update:` below that)